ethereum/pow/merkletree/src/main/java/tech/pegasys/teku/ethereum/pow/merkletree/DepositTree.java

```diff
@@ -94,12 +95,8 @@ public DepositTree getTreeAtDepositIndex(final long lastDepositIndex) {
     checkArgument(
         lastDepositIndex >= finalizedDepositCount,
         "Cannot recreate tree before finalized deposit count");
-    final DepositTree treeForProof;
-    if (finalizedExecutionBlock.isPresent()) {
-      treeForProof = fromSnapshot(getSnapshot());
-    } else {
-      treeForProof = new DepositTree();
-    }
+    final DepositTree treeForProof =
+        getSnapshot().map(DepositTree::fromSnapshot).orElse(new DepositTree());
```

Comment:
```suggestion
        getSnapshot().map(DepositTree::fromSnapshot).orElseGet(DepositTree::get);
```

to avoid always creating a new DepositTree even if the snapshot is available.

Reply:
Nice catch!